modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -413,6 +423,8 @@ private CompletableFuture<Void> calculateZoneAssignmentsAndCreateReplicationNode
         return inBusyLockAsync(busyLock, () -> {
             int zoneId = zoneDescriptor.id();
 
+            zoneResourcesManager.registerZonePartitionCount(zoneId, zoneDescriptor.partitions());
```

Comment:
I don't like such approach. 
What about using
`catalogMgr.activeCatalog(metaStorageMgr.timestampByRevisionLocally(revision).longValue()).zone(zonePartitionId.zoneId()).partitions();` in order to retrieve partitions count in `PartitionReplicaLifecycleManager#createZonePartitionReplicationNode` and thus propagating the count to ZoneResourcesManager#getOrCreatePartitionTxStateStorage
?

Reply:
1. The timestamp on MS entry for a catalog update is not the same as activation timestamp of the catalog update
2. This seems really fragile
3. We could probably get into trouble with MS compaction

Current approach is simple and solves the task at hand. Also, it doesn't introduce races. Why don't you like it?